packages/protocol/contracts/common/Accounts.sol

```diff
@@ -284,40 +352,196 @@ contract Accounts is
    * @dev v, r, s constitute `signer`'s signature on `msg.sender`.
    */
   function authorizeAttestationSigner(address signer, uint8 v, bytes32 r, bytes32 s) public {
-    Account storage account = accounts[msg.sender];
-    authorize(signer, v, r, s);
-    account.signers.attestation = signer;
+    authorizeSignerWithSignature(signer, AttestationSigner, v, r, s);
+    setDefaultSigner(signer, AttestationSigner);
+
     emit AttestationSignerAuthorized(msg.sender, signer);
   }
 
+  /**
+   * @notice Begin the process of authorizing an address to sign on behalf of the account
+   * @param signer The address of the signing key to authorize.
+   * @param role The role to authorize signing for.
+   */
+  function authorizeSigner(address signer, string memory role) public {
+    require(isAccount(msg.sender), "Unknown account");
+    require(
+      isNotAccount(signer) && isNotAuthorizedSignerForAnotherAccount(signer),
+      "Cannot re-authorize address signer"
+    );
+
+    accounts[msg.sender].signerAuthorizations[role][signer] = SignerAuthorization({
+      started: true,
+      completed: false
+    });
+
+    emit SignerAuthorized(msg.sender, signer, role);
+  }
+
+  /**
+   * @notice Finish the process of authorizing an address to sign on behalf of the account. 
+   * @param _account The address of account that authorized signing.
+   * @param role The role to finish authorizing for.
+   */
+  function completeSignerAuthorization(address _account, string memory role) public {
```

Comment:
maybe I'm not familiar enough with CIP10 but why separate `authorize` and `complete`, and why limit this to only `signer == msg.sender`?
what is the downside to someone authorizing and completing you as a signer on your behalf?

Reply:
It's a double opt-in. Like we have for our existing roles. You don't really want to allow the "burning" of an address in this scheme without their consent

Reply:
Agree that it's a little confusing. I guess there's two things at play here

- the act of proving you own an address
- setting that address as an authorised signer for an account.

Our system treats those two as one step currently. One question to ask is why have `authorizeSignerWithSignature` at all? The answer being we want to prove proof of possession before saying XXX was authorised by YYY. 

 Having `completeSignerAuthorization` is just the proof of possession for SC addresses.